### PR TITLE
Support for Nested structs

### LINKED ISSUE #4: Support for Nested Structures
Waiting on support for [Nested Structs](https://github.com/dart-lang/sdk/issues/37271) in `dart:ffi`
Currently, All members of a Struct are removed from bindings if it has another struct as a member

### PR DESCRIPTION
**Pull request by mannprerak2**
Closes #4

- support for nested structs
- Updated readme, changelog, test
- Minor changes to remove null usage from code.

### CODE REVIEW

**Comment by mannprerak2** [comment]
@jpnurmi hey sorry for making you do this again, will you please verify consent for @googlebot again.

**Comment by jpnurmi** [comment]
@googlebot I consent.

**Comment by jpnurmi** [comment]
No worries! :)

**Comment by mannprerak2** [comment]
Ahh, I made the PR on the wrong branch again.

File: .github/workflows/test-package.yml [hunk 1]
```diff
@@ -3,9 +3,9 @@ name: Dart CI
 on:
   # Run on PRs and pushes to the default branch.
   push:
-    branches: [ master ]
+    branches: [ dart-dev ]
```
Comment (dcharkes) [inline .github/workflows/test-package.yml:6]: also `[ master, dart-dev ]`

**Review by dcharkes — APPROVED** [review]

File: .github/workflows/test-package.yml (hunk 1 above)
Comment (mannprerak2) [reply .github/workflows/test-package.yml]: Done.

**Comment by dcharkes** [comment]
https://pub.dev/packages/ffigen/versions/2.0.0-dev.0 🎉

**Pull request merged**